dashboard/app/helpers/admin_reports_helper.rb

```diff
@@ -11,4 +11,36 @@ def get_gender_percentage(this_gender_count, other_gender_count)
   def get_percentage_from_ratio(ratio)
     return number_with_precision(100.0 * ratio, precision: 2).to_s + '%'
   end
+
+  # From a hash of hashes giving counts, get_cumulatives constructs a hash of hashes giving
+  # cumulative counts, e.g., transforms stats[key][subkey] = count to cumulatives[key][subkey] =
+  # cumulative_count, where cumulative_count := SUM(stats[key][i] : i <= subkey).
+  # The max_counts parameter is used to determine the space of subkeys, e.g.,
+  # cumulatives[key][subkey] will exist exactly when subkey <= max_counts[key].
+  def get_cumulatives(max_counts, stats)
+    # Construct the cumulatives hash manually. Since we ultimately want to display zero values, we
+    # explicitly initialize the hash.
+    cumulatives = Hash.new {|hash, key| hash[key] = {}}
+    max_counts.each do |key, max_count|
+      (0..max_count).each do |n|
+        cumulatives[key][n] = 0
+      end
+    end
```

Comment:
My belief is no. That would cause cumulatives[key][subkey] to be initialized to 0 if not already defined. But it would not exist (as I want it to) if never accessed in the loop below.